apps/src/aichat/views/AichatView.tsx

```diff
@@ -23,23 +26,36 @@ const AichatView: React.FunctionComponent = () => {
     dispatch(sendSuccessReport('aichat'));
   }, [dispatch]);
 
-  const initialAiCustomizations = useAppSelector(
+  const levelAiCustomizationsWithVisibility = useAppSelector(
     state =>
       (state.lab.levelProperties as AichatLevelProperties | undefined)
         ?.initialAiCustomizations || EMPTY_AI_CUSTOMIZATIONS
   );
 
+  const studentAiCustomizations = JSON.parse(
+    useAppSelector(
+      state => (state.lab.initialSources?.source as string) || '{}'
+    )
+  );
+
   useEffect(() => {
-    const aiCustomizations: AiCustomizations = {
-      botName: initialAiCustomizations.botName.value,
-      temperature: initialAiCustomizations.temperature.value,
-      systemPrompt: initialAiCustomizations.systemPrompt.value,
-      retrievalContexts: initialAiCustomizations.retrievalContexts.value,
-      modelCardInfo: initialAiCustomizations.modelCardInfo.value,
+    const levelAiCustomizations: AiCustomizations = {
+      botName: levelAiCustomizationsWithVisibility.botName.value,
+      temperature: levelAiCustomizationsWithVisibility.temperature.value,
+      systemPrompt: levelAiCustomizationsWithVisibility.systemPrompt.value,
+      retrievalContexts:
+        levelAiCustomizationsWithVisibility.retrievalContexts.value,
+      modelCardInfo: levelAiCustomizationsWithVisibility.modelCardInfo.value,
     };
 
-    dispatch(setAiCustomizations(aiCustomizations));
-  }, [dispatch, initialAiCustomizations]);
+    const reconciledAiCustomizations = Object.keys(studentAiCustomizations)
```

Comment:
hmm ok that makes sense. Is there an actual case where the levelbuilder and student customizations could diverge if the field is hidden or readonly, or just feels like a better practice? I thought the student's customizations would always just reflect the levelbuilder values, since they cannot change them.

Reply:
Yeah, one specific case would be if a readonly/hidden value is changed after the student runs the project for the first time. The student would have the old value saved to their main.json but we'd want to use the updated levelbuilder value

Reply:
Yeah that makes sense -- I think my question is whether that a thing that actually happens (ie, levelbuilders updating level config after a level is in use)? It feels potentially more confusing to me that a student might have written some customization on a level, then it partially (? if we're doing this per field) is overwritten by a levelbuilder?

Reply:
Yeah, I do think while it's uncommon it could definitely happen as things change in the pilot (ex. we learn that a certain system prompt isn't working well or providing bad responses). But to clarify, this would only apply to readonly/hidden fields, so the student would never be able to customize those fields in the first place. For editable fields, we'll always go with the latest student-edited value, even if the levelbuilder value has been updated. Here's a thread with Dan & Sam discussing this: https://codedotorg.slack.com/archives/C06FELVTV0Q/p1710190631660709

Reply:
Super helpful context, thank you!

Reply:
Updated, please take another look when you have a chance!